jax/_src/lax/lax.py

```diff
@@ -966,12 +966,46 @@ class GatherDimensionNumbers(NamedTuple):
   start_index_map: Sequence[int]
 
 
+class GatherScatterMode(Enum):
+  # Indices will be clamped to the nearest in-range value, i.e., such that the
+  # entire window to be gathered is in-range.
+  CLIP = 0
```

Comment:
Do you like using `auto`, as in `from enum import auto` and `CLIP = auto()` etc?

(My favorite way to write enums is like namedtuples, as in `GatherScatterMode = enum.Enum('GatherScatterMode', ['CLIP', 'FILL_OR_DROP', 'PROMISE_IN_BOUNDS'])`. But then we don't have space for the comments, or for adding the `from_any` method, so probably not the best option here. Just thought I'd share!)